packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -120,34 +154,68 @@ export class ServiceBuilderImpl implements ServiceBuilder {
         reject(e);
       });
 
-      const server = stoppable(
-        app.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      const useHttps = baseUrl.indexOf(DEFAULT_PROTOCOL) > -1;
+      let server: http.Server;
+
+      if (useHttps) {
+        const certificateAttributes: Array<any> = [];
+
+        Object.getOwnPropertyNames(
+          (certificateOptions?.attributes as any).data,
```

Comment:
This will very much not work :p. Config can have multiple layers of data and this will just look at the first one. What you're doing here can be done either with `.get(key)` or `.keys()` instead. Using `keys` is preferred since you'll end up getting nicer error messages for missing or bad values.

Reply:
It works on my machine. ;P To be honest, the config system is a bit confusing. It seems like some loaders have artificial dependencies (root package.json), Its uncertain when you have to load a function and/or when you load a reader/api. The configuration objects are not what they seem at runtime and since it all JS that will lead to people digging into the objects to find a way thru. I will rework this to use the .get and .keys methods. :)